test/test_optim.py

```diff
@@ -1291,6 +1301,70 @@ def closure():
             res2 = optim_neg_inf.step(closure)
             self.assertEqual(type(res1), type(res2))
 
+    @onlyCPU
+    @skipIfTorchDynamo("https://github.com/pytorch/pytorch/issues/123238")
```

Comment:
Let's also have this live in test_torch.py with its counterpart

Reply:
Keeping it in test_cuda.py is fine too--is there a reason to not do it? We should be able to add the OptimizerInfo stuff to it even in test_cuda.py, but you don't have to worry about that--there are other PRs ongoing at the moment that introducing the infra to these tests. For right now, it is fine to have the cpu and gpu tests be separate tests but they should live in one place (test_cuda)

Reply:
Thanks for suggesting. I agree to have both cpu test and gpu test. But we cannot put all these test into `test_cuda` now since `test_cuda` only works when we have `cuda`. 
```
if not TEST_CUDA:
    print('CUDA not available, skipping tests', file=sys.stderr)
    TestCase = NoTest  # noqa: F811
```
So I still keep in `test_optim.py` but followed your suggestion that `_test_grad_scaling_autocast_fused_optimizers ` is now controlled with `OptimizerInfo`